src/PostgREST/AppState.hs

```diff
@@ -46,7 +46,9 @@ import Protolude
 
 
 data AppState = AppState
-  { statePool         :: SQL.Pool -- | Connection pool, either a 'Connection' or a 'ConnectionError'
+  -- | Database connection pool; 'Nothing' indicates the pool has been destroyed
+  { statePool         :: IORef (Maybe SQL.Pool)
```

Comment:
This is all (i.e., `IORef (Maybe SQL.Pool)` as well as `usePool`, `destroyPool`, `flushPool` below) a *bit* messy. It might be nicer to hide all of that in a separate `PostgREST.Pool` module with

```
module PostgREST.Pool
   ( Pool
   , init
   , use
   , flush
   , destroy
   ) where

newtype Pool = Pool (IORef (Maybe SQL.Pool))

...
```

and then moving back to offering `AppState.getPool :: AppState -> Pool` and calling it via e.g. `Pool.use (AppState.getPool appState) xxx` all over the place.

That approach would also lend itself to unit testing our pool wrapper. On the other hand I'm not sure it's worth the effort right now so I decided ot file the PR as it is for the moment. What do you think?

Reply:
Sure, that's fine by me. The Pool module can be added later in another PR.